Tests/fixtures/bots.yml

```diff
@@ -3945,3 +3945,124 @@
+-
+  user_agent: Mozilla/5.0 (compatible; Cocolyzebot/1.0; https://cocolyze.com/bot)
+  bot:
+    name: Cocolyzebot
+    category: Crawler
+    url: https://cocolyze.com/en/cocolyzebot
+    producer:
+      name: VSI INNOVATION SAS
+      url: https://vsi-innovation.com/
+-
+  user_agent: veryhip (http://veryhip.com/)
+  bot:
+    name: VeryHip
+    category: Crawler
+    url: https://veryhip.com/
+    producer:
+      name: VeryHip
+      url: https://veryhip.com/
+-
+  user_agent: Mozilla/5.0 (compatible; LinkpadBot/2.3; +http://linkpad.org/robot/)
+  bot:
+    name: LinkpadBot
+    category: Crawler
+    url: https://www.linkpad.org/
+    producer:
+      name: Solomono LLC
+      url: https://www.linkpad.org/
+-
+  user_agent: Mozilla/4.0 (compatible; MuscatFerret/1.7; http://www.webtop.com/)
+  bot:
+    name: MuscatFerret
+    category: Crawler
+    url: http://www.webtop.com/
+-
+  user_agent: Quora-Bot/1.0 (http://www.quora.com)
+  bot:
+    name: Quora Bot
+    category: Crawler
+    url: ""
+    producer:
+      name: Quora
+      url: https://www.quora.com/
+-
+  user_agent: PageThing.com
+  bot:
+    name: PageThing
+    category: Crawler
+    url: https://www.pagething.com/
+    producer:
+      name: SPECIALNOISE LTD
+      url: https://www.specialnoise.com/
+-
+  user_agent: ArchiveBox/0.5.3 (+https://github.com/ArchiveBox/ArchiveBox/)
+  bot:
+    name: ArchiveBox
+    url: https://archivebox.io/
+    category: Crawler
+    producer:
+      name: ""
+      url: ""
+-
+  user_agent: Choosito/1.0 +http://www.choosito.com/
+  bot:
+    name: Choosito
+    url: https://www.choosito.com/
+    category: Crawler
+    producer:
+      name: Choosito! Inc.
+      url: https://www.choosito.com/
+-
+  user_agent: datagnionbot (+http://www.datagnion.com/bot.html)
+  bot:
+    name: datagnionbot
+    url: https://www.datagnion.com/bot.html
+    category: Crawler
+    producer:
+      name: DATAGNION GMBH
+      url: https://www.datagnion.com/
+-
+  user_agent: InterNaetBoten/0.99 (http://www.xn--internt-bxa.nu)
+  bot:
+    name: Generic Bot
+-
+  user_agent: EasyBib AutoCite (http://autocite-info.citation-api.com/)
+  bot:
+    name: Generic Bot
+-
+  user_agent: Bidtellect/0.0.643.0
+  bot:
+    name: Generic Bot
+-
+  user_agent: Mozilla/5.0 (compatible; meg/0.2; +https://github.com/tomnomnom/meg)
+  bot:
+    name: Generic Bot
+-
+  user_agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:66.0) WhatCMS/1.0
+  bot:
+    name: WhatCMS
+    url: https://whatcms.org/
+    category: Crawler
+    producer:
+      name: ""
+      url: ""
+-
+  user_agent: httpx - Open-source project (github.com/projectdiscovery/httpx)
+  bot:
+    name: httpx
+    url: https://github.com/projectdiscovery/httpx
+    category: Crawler
+    producer:
+      name: ""
+      url: ""
+-
+  user_agent: 'Expanse indexes the network perimeters of our customers. If you have any questions or concerns, please reach out to: scaninfo@expanseinc.com'
```

Comment:
strange useragent

Reply:
Seen it in my logs a few days ago.

Reply:
```
HuaweiWebCatBot/6.0) (To acquire the allowed html pages as reliable information of URL categorization in the automatic process for Huawei Web Categorization.; https://isecurity.huawei.com/; sec at huawei dot com)
```

I found a similar one, probably there is one teacher